arrows-stdlib-cats/src/main/scala/arrows/stdlib/cats.scala

```diff
@@ -0,0 +1,107 @@
+package arrows.stdlib
+
+import cats.{ Applicative, StackSafeMonad }
+import cats.effect.{ Async, ExitCase }
+import cats.arrow.ArrowChoice
+import cats.kernel.{ Monoid, Semigroup }
+import cats.mtl.ApplicativeAsk
+
+import scala.concurrent.Promise
+import scala.util.control.NonFatal
+import scala.util.{ Failure, Success }
+
+object Cats extends ArrowInstances
+
+sealed abstract class ArrowInstances extends ArrowInstances1 {
+
+  implicit def catsMonadForArrow[E]: Async[Arrow[E, ?]] = new Async[Arrow[E, ?]] with StackSafeMonad[Arrow[E, ?]] {
```

Comment:
No, that would constraint it unnecessary, An `Arrow[E, A]` is just as much valid as a `Task[A]`. Though for `Effect` it does need to be `Task`, as you can't implement `runAsync` otherwise. For `Async` we can do it for any `Arrow`.

Reply:
Arrow is more general then monad, so I don't think it's a good fit to a monad typeclass. Also, the fixed `Unit` input is important for performance. This has much better performance:

```scala
    def flatMap[A, B](fa: Task[A])(f: A => Task[B]): Task[E, B] =
      fa.flatMap(f)
```

than

```scala
    def flatMap[A, B](fa: Arrow[E, A])(f: A => Arrow[E, B]): Arrow[E, B] =
      Arrow[E].flatMap(e => fa(e).flatMap(a => f(a)(e)))
```

Reply:
Right yeah, that's why I need to write the `Effect` type for `Task`, but we should still provide all the instances that make sense :)

Reply:
For example, we also have `Async` instances for `Kleisli[F, R, A]` where `F` has an `Async` instance itself. Since `Arrow` is roughly equivalent to that, it stands to reason that we should provide that too :)